### DISCUSSION TITLE
Suggest adding in a shortcut to toggle sidebar hide show

### DISCUSSION THREAD

**Discussion by rr-tw5** [NicolasPetton/Notebook#23] [Ideas]
For those of us who are keyboard driven, that can be a great help.

I've made a toggle-notebook-sidebar series of shortcuts. Even though it's really obvious, here's the code.

Title: $:/core/ui/KeyboardShortcuts/toggle-notebook-sidebar

<$list filter="[[$:/state/notebook-sidebar]is[missing]] [{$:/state/notebook-sidebar}removeprefix[yes]]" emptyMessage="""
<$action-setfield $tiddler="$:/state/notebook-sidebar" text="yes"/>
""">
<$action-setfield $tiddler="$:/state/notebook-sidebar" text="no"/>
</$list>

Field:
key: ((toggle-notebook-sidebar))

Then I changed the default sidebar key  

But thinking aloud, as I wrote before about hiding the other sidebar (perhaps incorrectly as an issue––as I also wrote there I'm learning GitHub etiquette): perhaps adding a 'reveal' to $:/core/ui/KeyboardShortcuts/toggle-sidebar would be helpful to share the same code according which theme is presently selected.

This works for me, if you're happy to overwrite another element of the core with your theme…

Title: $:/core/ui/KeyboardShortcuts/toggle-sidebar 

<$reveal state="$:/theme" type="nomatch" text="$:/themes/nico/notebook">
<$list filter="[[$:/state/sidebar]is[missing]] [{$:/state/sidebar}removeprefix[yes]]" emptyMessage="""
<$action-setfield $tiddler="$:/state/sidebar" text="yes"/>
""">
<$action-setfield $tiddler="$:/state/sidebar" text="no"/>
</$list>
</$reveal>
<$reveal state="$:/theme" type="match" text="$:/themes/nico/notebook">
<$list filter="[[$:/state/notebook-sidebar]is[missing]] [{$:/state/notebook-sidebar}removeprefix[yes]]" emptyMessage="""
<$action-setfield $tiddler="$:/state/notebook-sidebar" text="yes"/>
""">
<$action-setfield $tiddler="$:/state/notebook-sidebar" text="no"/>
</$list>
</$reveal>

**Comment by NicolasPetton** [comment]
@rr-tw5 thanks for the suggestion.

I'll add a new tiddler for toggling the sidebar, without overriding `$:/core/ui/KeyboardShortcuts/toggle-sidebar`.

**Comment by NicolasPetton** [comment]
@rr-tw5 can you check commit c51b513 and tell me if that suits your needs?

**Comment by rr-tw5** [comment]
Many thanks.

The code looks good to me. 

I hadn't thought of that solution, so by sharing the 'key' field we can call more than one action from one keyboard shortcut? 
So both $:/state/notebook-sidebar and $:/state/sidebar will have their text updated on 'alt-shift-S'?

As I said I'm new to GitHub: How do you suggest I test it?

**Comment by rr-tw5** [comment]
So I just went ahead and created the tiddler manually. Although it works well in notebook, the shortcut no longer works in vanilla theme. I remember I noticed the same thing happens with search in sidebar 'ctrl-shift-F'.

If I'm right, then it would be better to write reveals into both/all the shortcut codes you've made, so that depending on the theme it acts appropriately (differently but consistently).

Seems once you override/duplicate a keyboard shortcut (either with the keystroke, or maybe also with the key field), it otherwise stops working.

So my suggestion is to add this code to the tiddler:


<$reveal state="$:/theme" type="nomatch" text="$:/themes/nico/notebook">
<$list filter="[[$:/state/sidebar]is[missing]] [{$:/state/sidebar}removeprefix[yes]]" emptyMessage="""
<$action-setfield $tiddler="$:/state/sidebar" text="yes"/>
""">
<$action-setfield $tiddler="$:/state/sidebar" text="no"/>
</$list>
</$reveal>
<$reveal state="$:/theme" type="match" text="$:/themes/nico/notebook">
<$list filter="[[$:/state/notebook-sidebar]is[missing]] [{$:/state/notebook-sidebar}removeprefix[yes]]" emptyMessage="""
<$action-setfield $tiddler="$:/state/notebook-sidebar" text="yes"/>
""">
<$action-setfield $tiddler="$:/state/notebook-sidebar" text="no"/>
</$list>
</$reveal>

**Reply by rr-tw5** [reply]
Okay, I've come to understand that all of this is probably because I changed the tiddlers from System to Shadows by editing them.

So what you've written is cleaner and more elegant. But (bear in mind, if you wish to) it won't protect you/notebook from people like me who open everything and seeing only a tick (don't wait for the hover to reveal the (x)) and save instead of cancel.

🤯

**Reply by NicolasPetton** [reply]
True, but that would be also the case for any shadow tiddler.

**Comment by rr-tw5** [comment]
One last comment (I hope).

I do find that there is an inability to use ctrl-shift-F for opening the search bar in both vanilla and notebook themes. Only the first one to use establish it seems to be able to continue to use it.

**Reply by NicolasPetton** [reply]
Hmm, that might be, I haven't looked at how kbd shortcuts are internally handled in TiddlyWiki. If you switch theme & reload, does it work as expected?

**Reply by rr-tw5** [reply]
By reload you mean after saving? I didn't try it on my wiki, I did it on https://nicolas.petton.fr/tw/notebook.html. 

I already have changed a lot more of the code on my instance. I expect you will be able to replicate it on your own instance of the notebook theme. Or maybe like Saq you don't actually use your own public offerings.

**Reply by NicolasPetton** [reply]
> Or maybe like Saq you don't actually use your own public offerings.

I'm not sure what you mean by that.

**Reply by rr-tw5** [reply]
Just that he wrote Streams which I use, but he doesn't personally use it.
